feature: Adicionando primeira versão do módulo de app_service_docker

chnacib opened the pull request:
Primeira versão do módulo que cria um app service que roda uma imagem docker



<!-- Por favor descreva seu pull request aqui. -->

- [X] Garanta que seu **topic/feature/bugfix branch** tenha uma branch nomeada e não a sua branch main esteja no PR
- [X] Dê um titulo que expresse o objetivo do PR
- [X] Associe seu PR a uma Issue criada no repositósito. Caso seja uma correção de linguagem ou pequenas correções, não é necessário
- [X] Descreva o objetivo do PR
- [X] Inclua links relevantes para a sua modificação/sugestão/correção
- [X] Descreva um passo-a-passo para testar o seu PR

## Issue

<!-- Link da issue -->

## Objetivo

Objetivo foi criar um app service para rodar uma imagem docker

## Referências

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service

## Como testar

[Link para instalar Azure-cli](https://docs.microsoft.com/pt-br/cli/azure/install-azure-cli)

```
cd /how-to-use-this-module
az login
terraform init
terraform plan
terraform apply
```

chnacib committed:
feature: Adicionando primeira versão do módulo de app_service_docker

Primeira versão do módulo que cria um app service que roda uma imagem docker

Co-authored-by: Marcos Rezende <rehzende@hotmail.com>
Co-authored-by: Carlos Oliveira <cdaniel.oliveira@outlook.com>
Co-authored-by: Antonio Fernandes <afernandescneto@gmail.com>
Co-authored-by: Paulo Henrique Conte <paulohenriqueconte@gmail.com>
Co-authored-by: Mauro Felipe <maurofelipe@gmail.com>
Co-authored-by: Luiz Aoqui <lgfa29@gmail.com>

lgfa29 committed:
chore: terraform fmt e mudar versão requirida

gomex commented:
O PR está falhando no pipeline de checagem do tfsec.

lgfa29 commented:
Outra coisa também é que semana passada saiu a versão 3.0.0 do provider da Azure e o resource `azurerm_app_service_plan` que a gente usou foi depreciado (https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service). Acho que seria bom a gente já começar usando o novo 🙂

lgfa29 committed:
feat: atualiza recursos utilizados para v3.0.0

A versão 3.0.0 do provider da Azure depreciou alguns recursos, inclusive
dois que estavam sendo utilizados nesse módulo: o azure_app_service e o
azure_app_service_plan.

Esse commit atualiza o módulo para utilizar os recursos novos que
substituiram os anteriores e também atualiza os parâmetros e variáveis
necessárias.

Co-authored-by: João Marcos de A. Dantas <joaomarcos.ad@gmail.com>
Co-authored-by: Mauro Felipe <maurofelipe@gmail.com>
Co-authored-by: Glenon Mateus <glenonmateus@gmail.com>
Co-authored-by: Carlos Henrique <carlos_henrique190@yahoo.com.br>
Co-authored-by: Anderson Lima <msgli34@gmail.com>
Co-authored-by: Lucas Nascimento <lucasnascimento.contas@outlook.com>

lgfa29 committed:
ci: Ignorando regra GEN001 do tfsec

Essa regra verifica que nenhum valor default é utilizado em variáveis
marcadas como sensitive. Mas como estamos utilizando uma string vazia
como default, não há problema de vazar dados sensíveis.

Co-authored-by: João Marcos de A. Dantas <joaomarcos.ad@gmail.com>
Co-authored-by: Mauro Felipe <maurofelipe@gmail.com>
Co-authored-by: Glenon Mateus <glenonmateus@gmail.com>
Co-authored-by: Carlos Henrique <carlos_henrique190@yahoo.com.br>
Co-authored-by: Anderson Lima <msgli34@gmail.com>
Co-authored-by: Lucas Nascimento <lucasnascimento.contas@outlook.com>
